src/go/k8s/apis/redpanda/v1alpha1/redpanda_clusterspec_types.go

```diff
@@ -0,0 +1,340 @@
+package v1alpha1
+
+import (
+	"encoding/json"
+)
+
+// RedpandaClusterSpec defines the desired state of Redpanda Cluster
+type RedpandaClusterSpec struct {
```

Comment:
For CRD conversion, I thought we would have to keep the CRD name and use different version such as v1beta1 instead of v1alpha1.

https://book.kubebuilder.io/multiversion-tutorial/conversion.html